usaspending_api/common/recipient_lookups.py

```diff
@@ -19,19 +28,27 @@ def obtain_recipient_uri(recipient_name, recipient_unique_id, parent_recipient_u
     Return example string: 11fcdf15-3490-cdad-3df4-3b410f3d9b20-C
 
     """
-    if (is_parent_recipient and not recipient_unique_id) or not (recipient_unique_id or recipient_name):
+    # Checks for two cases to return early:
+    # - Parent recipient requires a unique identifier (UEI or DUNS)
+    # - When all components of the Recipient Hash are NULL (UEI, DUNS, or Name)
+    if (recipient_uei is None and recipient_unique_id is None) and (is_parent_recipient or recipient_name is None):
         return None
 
-    if recipient_unique_id:
-        recipient_hash = fetch_recipient_hash_using_duns(recipient_unique_id)
+    if recipient_uei is not None:
+        recipient = RecipientProfile.objects.filter(uei=recipient_uei).values("recipient_hash").first()
+    elif recipient_unique_id is not None:
+        recipient = RecipientLookup.objects.filter(duns=recipient_unique_id).values("recipient_hash").first()
     else:
-        recipient_hash = None
+        recipient = None
+    recipient_hash = recipient.get("recipient_hash") if recipient else None
```

Comment:
What exactly are you doing on this line since you set `recipient` to `None` on the preceding line?